examples/statistics/time_series_histogram.py

```diff
@@ -0,0 +1,86 @@
+"""
+=====================
+Time Series Histogram
+=====================
+
+This example demonstrates how to efficiently visualize large numbers of time
+series in a way that could potentially reveal hidden substructure and patterns
+that are not immediately obvious.
+
+The first plot shows the typical way of visualizing multiple time series by
+overlaying them on top of each other with ``plt.plot``. The second and third
+plots show how to reinterpret the data as a 2d histogram.
+"""
+from copy import copy
+import time
+
+import numpy as np
+import numpy.matlib
+import matplotlib.pyplot as plt
+from matplotlib.colors import LogNorm
+
+_, axes = plt.subplots(nrows=3, figsize=(10, 6 * 3))
+
+# Make some data; a 1D random walk + small fraction of sine waves
+num_series = 10000
```

Comment:
This example takes over 40 seconds to run, out of a 15 minute process.  It needs to be much faster to be worth it and I'm sure the effect can be achieved with fewer than 10,000 time series.

Reply:
Does the CI pipeline run over the same code multiple times or something? On my machine it only takes 5 seconds to run! I'll take a look and see if I can reduce the number of series while still illustrating the other advantages.

Reply:
Just look at the output on CI - it says the time...

Reply:
Reducing the number of time series from 10k -> 1k also reduces the time it takes (on my machine) from ~4 secs to ~0.3 secs, but now the sinusoidal signal is clearly visible in the top plot, which kind of takes away from the advantage of using the histogram:
![asdf](https://user-images.githubusercontent.com/10482181/96651588-1ef2f280-1303-11eb-9c14-53eccef3820a.png)

Reply:
I can increase `alpha=0.1` to obfuscate it again, but this kind of feels like "cheating".
![asdf](https://user-images.githubusercontent.com/10482181/96651924-d5ef6e00-1303-11eb-97f8-482d7e50b353.png)

Reply:
Sorry I mistyped at 40 s.  It is 14s: https://47944-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/statistics/time_series_histogram.html#sphx-glr-gallery-statistics-time-series-histogram-py

Reply:
What do you think of the above example(s)? The fact that you can see the sinusoid now when using `plot` kind of detracts from the proposed advantages of increased visibility, but the histograms definitely look "sexier" with the `plasma` color coding IMO.

Reply:
I think thats fine.  I think its also reasonable to increase the alpha.  That alpha effect is pretty fragile at such a low value.

Reply:
Ok, so CI now says [it only takes 2.4 seconds](https://48080-1385122-gh.circle-artifacts.com/0/doc/build/html/gallery/statistics/time_series_histogram.html#sphx-glr-gallery-statistics-time-series-histogram-py) after reducing the number of series by a factor of 10 (not quite linear scaling unfortunately). Does that seem low enough, or should we try and reduce the build time a bit more?